src/main/java/com/fxgraph/edges/CorneredLoopEdge.java

```diff
@@ -0,0 +1,215 @@
+	}
+
+	public DoubleProperty widthProperty() {
+		return widthProperty;
+	}
+
+	public enum Position {
+		TOP, RIGHT, BOTTOM, LEFT;
+
+		boolean isVertical() {
+			return this == LEFT || this == RIGHT;
+		}
+
+		boolean isHorizontal() {
+			return this == TOP || this == BOTTOM;
+		}
+	}
+
+	public static class EdgeGraphic extends AbstractEdgeGraphic {
+
+		private final DoubleBinding sourceX;
+		private final DoubleBinding sourceY;
+		private final DoubleBinding targetX;
+		private final DoubleBinding targetY;
+		private final DoubleBinding centerX;
+		private final DoubleBinding centerY;
+		private final Line lineA = new Line();
+		private final Line lineB = new Line();
+		private final Line lineC = new Line();
+
+		public EdgeGraphic(Graph graph, CorneredLoopEdge edge, Position position, StringProperty textProperty,
+						   DoubleProperty separation, DoubleProperty width) {
+			DoubleBinding w = edge.getTarget().getWidth(graph).add(0.000001);
+			DoubleBinding h = edge.getTarget().getHeight(graph).add(0.000001);
+			DoubleBinding x = edge.getTarget().getXAnchor(graph);
+			DoubleBinding y = edge.getTarget().getYAnchor(graph);
+
+			if (position.isVertical()) {
+				sourceY = y.subtract(h.divide(2.0).multiply(width));
+				targetY = y.add(h.divide(2.0).multiply(width));
+
+				if (position == Position.LEFT) {
+					sourceX = x.subtract(w.divide(2.0));
+					targetX = sourceX;
+					centerX = sourceX.subtract(w.divide(2.0)).subtract(separation);
+				} else {
+					sourceX = x.add(w.divide(2.0));
+					targetX = sourceX;
+					centerX = sourceX.add(w.divide(2.0)).add(separation);
+				}
+
+				centerY = sourceY.add(targetY).divide(2);
+			} else {
+				sourceX = x.subtract(w.divide(2.0).multiply(width));
+				targetX = x.add(w.divide(2.0).multiply(width));
+
+				if (position == Position.TOP) {
+					sourceY = y.subtract(h.divide(2.0));
+					targetY = sourceY;
+					centerY = sourceY.subtract(h.divide(2.0)).subtract(separation);
+				} else {
+					sourceY = y.add(h.divide(2.0));
+					targetY = sourceY;
+					centerY = sourceY.add(h.divide(2.0)).add(separation);
+				}
+
+				centerX = sourceX.add(targetX).divide(2);
+			}
+
+			text.textProperty().bind(textProperty);
+			text.getStyleClass().add("edge-text");
+			final DoubleProperty textWidth = new SimpleDoubleProperty();
+			final DoubleProperty textHeight = new SimpleDoubleProperty();
+			final Runnable recalculateWidth = () -> {
+				textWidth.set(text.getLayoutBounds().getWidth());
+				textHeight.set(text.getLayoutBounds().getHeight());
+			};
+			text.parentProperty().addListener((obs, oldVal, newVal) -> recalculateWidth.run());
+			text.textProperty().addListener((obs, oldVal, newVal) -> recalculateWidth.run());
+
+			if (position == Position.TOP) {
+				text.yProperty().bind(centerY.subtract(textHeight.divide(2)));
+			} else {
+				text.yProperty().bind(centerY.add(textHeight.divide(2)));
+			}
+			text.xProperty().bind(centerX.subtract(textWidth.divide(2)));
```

Comment:
The y positions for top/bottom can be made to fit better. 

And for left/right loops they should be centered. I'll fix this tomorrow.